src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php

```diff
@@ -137,6 +138,17 @@ private function createSchema(ObjectManager $em)
         ));
     }
 
+    public function resultTypesProvider()
+    {
+        $entity = new SingleIntIdEntity(1, 'foo');
+
+        return array(
+            array($entity, array($entity)),
+            array($entity, new \ArrayIterator(array($entity))),
+            array($entity, new ArrayCollection(array($entity))),
+        );
+    }
```

Comment:
I looked existing code - not so simple, there are different tests in different approaches. But I agree that the data provider close to the test easier to read. Fixed.